heareval/train_baseline.py

```diff
@@ -0,0 +1,164 @@
+"""
+Train the baseline model on a single multiclass task.
+
+TODO: Some of the boilerplate code is shared with task_embeddings.py
+and can be cleaned up.
+"""
+
+import os.path
+from importlib import import_module
+
+import numpy as np
+import pytorch_lightning as pl
+import soundfile as sf
+import torch
+from torch.utils.data import DataLoader, random_split
+
+import heareval.baseline
+
+from csvdataset import CSVDataset
+from heareval.baseline import RandomProjectionMelEmbedding
+
+
+TASK = "coughvid-v2.0.0"
+
+# For this task
+# TODO: Don't hardcode
+FRAME_RATE = 8
+SAMPLE_LENGTH_SECONDS = 8.0
+N_LABELS = 3
+
+# TODO: Support for multiple GPUs?
+device = "gpu" if torch.cuda.is_available() else "cpu"
+
+# +
+VALIDATION = 0.2
+
+AUDIO_BATCH_SIZE = 64
+FRAME_BATCH_SIZE = 64
+EMBEDDING_SIZE = 4096
+
+# There should be a way to use this code both for fine-tuning and non-fined-prediction
+FINE_TUNE = True
+
+# +
+# TODO: Use validation if it exists
+# Not sure why I have to add .. to the path
+full_train_dataset = CSVDataset(
+    os.path.join("../tasks", TASK, f"train.csv"), labels_as_ints=True
+)
+n_validation = int(round(len(full_train_dataset) * VALIDATION))
+train_dataset, validation_dataset = random_split(
+    full_train_dataset,
+    [len(full_train_dataset) - n_validation, n_validation],
+    generator=torch.Generator().manual_seed(42),
+)
+test_dataset = CSVDataset(
+    os.path.join("../tasks", TASK, f"test.csv"), labels_as_ints=True
+)
+
+train_dataloader = DataLoader(
+    train_dataset, batch_size=AUDIO_BATCH_SIZE, shuffle=True, drop_last=False
+)
+validation_dataloader = DataLoader(
+    validation_dataset, batch_size=AUDIO_BATCH_SIZE, shuffle=False, drop_last=False
+)
+test_dataloader = DataLoader(
+    test_dataset, batch_size=AUDIO_BATCH_SIZE, shuffle=False, drop_last=False
+)
+
+# -
+
+
+class MulticlassEmbedding(pl.LightningModule):
+    SR = heareval.baseline.input_sample_rate()
+
+    def __init__(self):
+        super().__init__()
+        self.embedding = RandomProjectionMelEmbedding()
+        # Add one because the start and end are framed
+        self.embedding_dim = int(
+            (FRAME_RATE * SAMPLE_LENGTH_SECONDS + 1) * EMBEDDING_SIZE
+        )
+        self.fc = torch.nn.Linear(self.embedding_dim, N_LABELS)
+        self.loss = torch.nn.CrossEntropyLoss()
+
+    def embeddings_from_filenames(self, filenames, split):
+        # Might want to make this an option in CSVDataset
+        audio = []
+        for f in filenames:
+            x, sr = sf.read(
+                os.path.join(os.path.join("../tasks", TASK, str(self.SR), split, f)),
+                dtype=np.float32,
+            )
+            assert sr == self.SR
+            audio.append(x)
+        audio = torch.tensor(np.vstack(audio), device=device)
+
```

Comment:
This is pretty gross and suggests maybe we should change the API more.

I am not sure if we call heareval.baseline.get_audio_embedding() if gradients for the model will be accumulated